thrift/src/test/java/com/linecorp/armeria/it/metrics/PrometheusMetricsIntegrationTest.java

```diff
@@ -0,0 +1,245 @@
+                     .aggregate().get();
+    }
+
+    private static Map<MyMetricLabel, String> defaultMetricName(RequestLog log,
+                                                                String serviceName) {
+        final RequestContext ctx = log.context();
+        final Object requestEnvelope = log.requestEnvelope();
+        final Object requestContent = log.requestContent();
+
+        final String path;
+        final String methodName;
+
+        if (requestContent instanceof RpcRequest) {
+            methodName = ((RpcRequest) requestContent).method();
+        } else if (requestEnvelope instanceof HttpHeaders) {
+            methodName = ((HttpHeaders) requestEnvelope).method().name();
+        } else {
+            methodName = log.method();
+        }
+
+        if (requestEnvelope instanceof HttpHeaders) {
+            path = ctx.path();
+        } else {
+            path = null;
+        }
+
+        return ImmutableMap.<MyMetricLabel, String>builder()
+                .put(MyMetricLabel.handler, serviceName)
+                .put(MyMetricLabel.path, firstNonNull(path, "__UNKNOWN_PATH__"))
+                .put(MyMetricLabel.method, firstNonNull(methodName, "__UNKNOWN_METHOD__"))
+                .build();
+    }
+
+    private enum MyMetricLabel implements MetricLabel<MyMetricLabel> {
+        path,
+        handler,
+        method,
+    }
+
+    @Test(timeout = 10000L)
+    public void normal() throws Exception {
+
+        makeRequest("world");
+        makeRequest("world");
+        assertThatThrownBy(() -> makeRequest("space"));
+        makeRequest("world");
+        assertThatThrownBy(() -> makeRequest("space"));
+        assertThatThrownBy(() -> makeRequest("space"));
+        makeRequest("world");
+
+        //Wait until all RequestLogs are collected.
+        given().ignoreExceptions().untilAsserted(() -> assertThat(makeMetricsRequest().content().toStringUtf8())
+                .contains("armeria_server_timer_count",
+                          "armeria_server_responseBytes_count",
+                          "armeria_server_requestBytes_count",
+                          "armeria_client_timer_count",
+                          "armeria_client_responseBytes_count",
+                          "armeria_client_requestBytes_count",
+                          "armeria_server_failure",
+                          "armeria_client_failure",
+                          "armeria_server_activeRequests",
+                          "armeria_client_activeRequests"));
+
+        final String content = makeMetricsRequest().content().toStringUtf8();
+        final String helloServiceMethod = "{path=\"/thrift\",handler=\"HelloService\",method=\"hello\",}";
+
+        //Server entry count check
+        assertThat(new BufferedReader(new StringReader(content))
+                           .lines()
+                           .map(String::trim)
+                           .filter(line -> line.startsWith("armeria_server_timer_count"))
```

Comment:
I think the assertions can be made more compact and precise using regex, what do you think?

```java
assertThat(content).containsPattern(Pattern.compile("^armeria_server_timer_count{path=\"/thrift\",handler=\"HelloService\",method=\"hello\",}=7.0$", MULTILINE));
```

Alternatively, if you stick with splitting and filtering, at least just do the split once above, and don't need the readers
```java
List<String> lines = Splitter.on('\n').trimResults().splitToList(content);
```